core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -26,80 +32,371 @@ object GraphQLActionBuilder {
-            }
         }
-
+        return tables
     }
 
     private fun handleOperation(
             actionCluster: MutableMap<String, Action>,
-            tableName: String?,
-            tableField: String?,
-            tableType: String?
+            methodName: String?,
+            methodType: String?,
+            tableType: String,
+            kindOfTableType: String,
+            kindOfTableField: String?,
+            table:List<Table>,
+            tableName: String,
+            IskindOfTableTypeOptional: Boolean,
+            IskindOfkindOfTableFieldOptional : Boolean
     ) {
+        if(methodName == null){
+            //TODO log warn
+            return;
+        }
+        if(methodType == null){
+            //TODO log warn
+            return;
+        }
+        val type = when{
+            methodType.equals("QUERY",true) -> GQMethodType.QUERY
+            methodType.equals("MUTATION",true) -> GQMethodType.MUTATION
+            else -> {
+                //TODO log warn
+                return
+            }
+        }
+
+        val actionId = "$methodName${idGenerator.incrementAndGet()}"
 
+        //TODO populate
 
-        val actionId = "$tableName$tableField${idGenerator.incrementAndGet()}"
-        val action = GraphQLAction(actionId, tableName, tableField, tableType)
+        val params = extractParams(methodName, tableType, kindOfTableType, kindOfTableField, table, tableName, IskindOfTableTypeOptional, IskindOfkindOfTableFieldOptional)
+
+        val action = GraphQLAction(actionId, methodName, type, params)
 
         actionCluster[action.getName()] = action
     }
+    private fun extractParams(
+            methodName: String,
+            tableType: String,
+            kindOfTableType: String,
+            kindOfTableField: String?,
+            table: List<Table>,
+            tableName: String,
+            IskindOfTableTypeOptional: Boolean,
+            IskindOfkindOfTableFieldOptional : Boolean
+    ): MutableList<Param> {
+
+        val params = mutableListOf<Param>()
+        val history: Deque<String> = ArrayDeque<String>()
+
+        val gene = getGene(tableType, kindOfTableField, kindOfTableType, table, tableName, history,IskindOfTableTypeOptional,IskindOfkindOfTableFieldOptional)
+
+            params.add(GQReturnParam(tableType, gene))
+
+        return params
+    }
+
+    private fun getGene(
+            tableType: String,
+            kindOfTableField: String?,
+            kindOfTableType: String,
+            table: List<Table>,
+            tableName: String,
+            history: Deque<String> = ArrayDeque<String>(),
+            IskindOfTableTypeOptional: Boolean,
+            IskindOfkindOfTableFieldOptional : Boolean
+    ): Gene {
+
+        when (kindOfTableField?.toLowerCase()) {
+            "list" -> {
+                if(IskindOfkindOfTableFieldOptional== true){
```

Comment:
as a non-nullable boolean, can use it directly, no need for the `==true`

Reply:
Done